Explorer/Assets/Scripts/ECS/SceneLifeCycle/Systems/LoadFixedPointersSystem.cs

```diff
@@ -62,25 +69,52 @@ private void ResolvePromises(ref FixedScenePointers fixedScenePointers, ref Proc
 
             fixedScenePointers.AllPromisesResolved = true;
 
-            for (var i = 0; i < fixedScenePointers.Promises.Length; i++)
+            if (fixedScenePointers.URNScenePromises is { Length: > 0 })
             {
-                ref AssetPromise<SceneEntityDefinition, GetSceneDefinition> promise = ref fixedScenePointers.Promises[i];
-                if (promise.IsConsumed) continue;
+                for (var i = 0; i < fixedScenePointers.URNScenePromises.Length; i++)
+                {
+                    ref AssetPromise<SceneEntityDefinition, GetSceneDefinition> promise = ref fixedScenePointers.URNScenePromises[i];
+                    if (promise.IsConsumed) continue;
+
+                    if (promise.TryConsume(World, out StreamableLoadingResult<SceneEntityDefinition> result))
+                    {
+                        if (result.Succeeded)
+                        {
+                            CreateSceneEntity(result.Asset, promise.LoadingIntention.IpfsPath);
+                            IReadOnlyList<Vector2Int> parcels = result.Asset.metadata.scene.DecodedParcels;
 
-                if (promise.TryConsume(World, out StreamableLoadingResult<SceneEntityDefinition> result))
+                            for (var j = 0; j < parcels.Count; j++)
+                                processedScenePointers.Value.Add(parcels[j].ToInt2());
+                        }
+                    }
+                    else
+                    {
+                        // at least one unresolved promises
+                        fixedScenePointers.AllPromisesResolved = false;
+                    }
+                }
+            }
+
+            if (fixedScenePointers.PointerScenesPromise is { IsConsumed: false })
+            {
+                if (fixedScenePointers.PointerScenesPromise.Value.TryConsume(World, out StreamableLoadingResult<SceneDefinitions> result))
```

Comment:
Typical mistake: you call `TryConsume` on `Nullable.Value`, thus you modify a copy, not the original value, so the next frame it will be in the unconsumed state

Reply:
I hadn't thought of that, good catch.

I removed the nullables usage, tackled at: https://github.com/decentraland/unity-explorer/pull/2229/commits/84d17fc187c0798859f7ace5fc638901b4f2e9c7

I also saw we are doing the same copy `TryConsume()` in [this other system](https://github.com/decentraland/unity-explorer/blob/075845e7d50193438fc63daf4353b2b14cc3d24f/Explorer/Assets/DCL/MapPins/Systems/MapPinLoaderSystem.cs#L88), I'll notify Davide to update it.